wallet/chain/p/builder/builder.go

```diff
@@ -1053,6 +1062,63 @@ func (b *builder) NewIncreaseBalanceTx(
 	return tx, b.initCtx(tx)
 }
 
+func (b *builder) NewDisableSubnetValidatorTx(
+	validationID ids.ID,
+	options ...common.Option,
+) (*txs.DisableSubnetValidatorTx, error) {
+	var (
+		toBurn  = map[ids.ID]uint64{}
+		toStake = map[ids.ID]uint64{}
+		ops     = common.NewOptions(options)
+	)
+	disableAuth, err := b.authorize(validationID, ops)
+	if err != nil {
+		return nil, err
+	}
+
+	memo := ops.Memo()
+	memoComplexity := gas.Dimensions{
+		gas.Bandwidth: uint64(len(memo)),
+	}
+	authComplexity, err := fee.AuthComplexity(disableAuth)
+	if err != nil {
+		return nil, err
+	}
+
+	complexity, err := fee.IntrinsicDisableSubnetValidatorTxComplexities.Add(
```

Comment:
this is safe because the receiver is not a pointer, right?

Reply:
correct

Reply:
yes